R/DataModel.R

```diff
@@ -391,58 +435,102 @@ uploadResults <- function(connectionDetails = NULL,
                                 pos,
                                 " through ",
                                 pos + nrow(chunk) - 1)
 
-        chunk <- checkAndFixColumnNames(
-          table = chunk,
-          tableName = env$specTableName,
-          zipFileName = zipFileName,
-          specifications = specifications
-        )
-        chunk <- checkAndFixDataTypes(
-          table = chunk,
-          tableName = env$specTableName,
-          zipFileName = zipFileName,
-          specifications = specifications
-        )
-        chunk <- checkAndFixDuplicateRows(
-          table = chunk,
-          tableName = env$specTableName,
-          zipFileName = zipFileName,
-          specifications = specifications
-        )
+        # Ensure all column names are in lowercase
+        colnames(chunk) <- tolower(colnames(chunk))
 
-        # Primary key fields cannot be NULL, so for some tables convert NAs to empty or zero:
-        toEmpty <- specifications %>%
-          dplyr::filter(
-            tableName == env$specTableName &
-              tolower(emptyIsNa) != "yes" &
-              grepl("varchar", dataType)
+        if (runCheckAndFixCommands) {
+          chunk <- checkAndFixColumnNames(
+            table = chunk,
+            tableName = env$specTableName,
+            zipFileName = zipFileName,
+            specifications = specifications
+          )
+          chunk <- checkAndFixDataTypes(
+            table = chunk,
+            tableName = env$specTableName,
+            zipFileName = zipFileName,
+            specifications = specifications
+          )
+          chunk <- checkAndFixDuplicateRows(
+            table = chunk,
+            tableName = env$specTableName,
+            zipFileName = zipFileName,
+            specifications = specifications
+          )
+
+          # Primary key fields cannot be NULL, so for some tables convert NAs to empty or zero:
+          toEmpty <- specifications %>%
+            dplyr::filter(
+              tableName == env$specTableName &
+                tolower(emptyIsNa) != "yes" &
+                grepl("varchar", dataType)
+            ) %>%
+            dplyr::select("columnName") %>%
+            dplyr::pull()
+          if (length(toEmpty) > 0) {
+            chunk <- chunk %>%
+              dplyr::mutate_at(toEmpty, naToEmpty)
+          }
+
+          toZero <- specifications %>%
+            dplyr::filter(
+              tableName == env$specTableName &
+                tolower(emptyIsNa) != "yes" &
+                dataType %in% c("int", "bigint", "float")
+            ) %>%
+            dplyr::select("columnName") %>%
+            dplyr::pull()
+          if (length(toZero) > 0) {
+            chunk <- chunk %>%
+              dplyr::mutate_at(toZero, naToZero)
+          }
+        }
+
+        # Ensure dates are formatted properly
+        toDate <- specifications %>%
+          filter(
+            .data$tableName == env$tableName &
```

Comment:
you need to add `dplyr::` namespacing here.

Also the use of `.data` in select statements (but not other functions like mutate) was deprecated for some reason (tidyverse rules...) so the build will now complain about this whenever you do it.